library/core/src/ptr/non_null.rs

```diff
@@ -1803,4 +1811,54 @@ mod verify {
         let maybe_null_ptr =  if kani::any() { xptr as *mut i32 } else { null_mut() };
         let _ = NonNull::new(maybe_null_ptr);
     }
+
+    #[kani::proof_for_contract(NonNull::replace)]
+    pub fn non_null_check_replace() {
+        let mut x: i32 = kani::any();
+        let mut y: i32 = kani::any();
+        kani::assume((&y as *const i32) != core::ptr::null());
+
+        if let Some(origin_ptr) = NonNull::new(&mut x as *mut i32) {
```

Comment:
The address of `x` is guaranteed to be non-NULL, so the result of `new` should be unwrapped.

Reply:
The code I'm referring to does:
```rust
NonNull::new(&mut x as *mut i32) {
```
which is guaranteed not to fail. The one that is failing does:
```rust
NonNull::new(x as *mut i32).unwrap();
```
which is different. Is there a reason for the change?